### PR TITLE
autotest: retrieve sim_time from lots of different messages

### PR DESCRIPTION
**Pull request by peterbarker**
Based on tridge's work here: https://github.com/ArduPilot/ardupilot/pull/15281/files

### CODE REVIEW

**Review by khancyr — CHANGES_REQUESTED** [review]
doesn't work as our current pattern is 
````
        tstart = self.get_sim_time()
        while True:
            self.wait_heartbeat()
            tdelta = self.get_sim_time_cached() - tstart
````
so we never have update ... unless we put a hook somewhere to always update self.mav.autotest_uptime for each message with time info ...

**Review by khancyr — APPROVED** [review]
@peterbarker that seems good ? any thing that should be tested ?

File: Tools/autotest/common.py [hunk 1]
```diff
@@ -2311,6 +2311,16 @@ def write_msg_to_tlog(self, msg):
         content = bytearray(struct.pack('>Q', usec) + msg.get_msgbuf())
         self.tlog.write(content)
 
+    def message_hook_timestamp_updater(self, mav, msg):
+        """Called as each mavlink msg is received."""
+        # swiped from mavutil:
+        if 'usec' in msg.__dict__:
```
Comment (khancyr) [inline Tools/autotest/common.py:2317]: shouldn't we check self.mav.sysid here too ?

**Comment by peterbarker** [comment]
It turns out that it is a far better idea to poll the ardupilot instance for the relevant message than to get timestamps from many messages; you can get a response to your poll in a very, very short period of time and not have to wait for the streams to kick in and send you a relevant message.

Making that change, however, shows up many issues - so that's something to do another day.

**Pull request closed without merging**